url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/XScriptEngine.java

```diff
@@ -0,0 +1,909 @@
+package xscript;
+
+import java.io.ByteArrayInputStream;
+import java.io.Externalizable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.ObjectStreamException;
+import java.io.PrintStream;
+import java.io.Reader;
+import java.io.Serializable;
+import java.io.StringReader;
+import java.lang.reflect.Proxy;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+import javax.script.AbstractScriptEngine;
+import javax.script.Bindings;
+import javax.script.Invocable;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngineFactory;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+import javax.tools.Diagnostic;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.DiagnosticCollector;
+import javax.tools.DiagnosticListener;
+
+import xscript.XExec.State;
+import xscript.compiler.XCompiler;
+import xscript.compiler.XFileReader;
+import xscript.compiler.XInternCompiler;
+import xscript.object.XConstPoolImpl;
+import xscript.object.XFunctionData;
+import xscript.object.XObject;
+import xscript.object.XObjectDataFunc;
+import xscript.object.XRuntime;
+import xscript.object.XTypeData;
+import xscript.object.XTypeDataBool;
+import xscript.object.XTypeDataConstPool;
+import xscript.object.XTypeDataFloat;
+import xscript.object.XTypeDataFunc;
+import xscript.object.XTypeDataInt;
+import xscript.object.XTypeDataList;
+import xscript.object.XTypeDataMap;
+import xscript.object.XTypeDataModule;
+import xscript.object.XTypeDataNull;
+import xscript.object.XTypeDataNumber;
+import xscript.object.XTypeDataString;
+import xscript.object.XTypeDataTuple;
+import xscript.object.XTypeDataWeakRef;
+import xscript.values.XValue;
+import xscript.values.XValueAccess;
+import xscript.values.XValueNull;
+import xscript.values.XValueObj;
+
+public class XScriptEngine extends AbstractScriptEngine implements Invocable,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 72 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.